_posts/2018/2018-04-29-use-hash-fetch-when-using-params-in-controllers.md

```diff
@@ -0,0 +1,77 @@
+---
+title: 'Use Hash#fetch when using Rails params in controllers'
+description: ''
+layout: article
+category: ruby
+image:
+  base: '2018/use-hash-fetch-when-using-params-in-controllers'
+  alt: 'Retriever'
+  source: 'https://unsplash.com/photos/Pzu9f6Nby5w'
+  credit: 'Mitchell Orr'
+---
+
+We often need to access a parameter from the URL that isn't part of the regular rails routes. Perhaps a query parameter such as `/search?q=term`.
+
+The parameters for a Rails action are stored in an `ActionController::Parameters` object which behaves quite a bit like a standard ruby hash.
+
+A ruby hash is typically accessed using square brackets, like `hash[key]`, but you can also use the method `fetch`, like `hash.fetch(key)`.
+
+
+## Instead of…
+
+...accessing required parameters from the `params` hash using the typical `#[]` method...
+
+```ruby
+class ThingController < ApplicationController
+  # ... actions
+
+  private
+
+   def an_important_param
+     params[:important_and_required]
+   end
+end
+```
+
+
+## Use…
+
+...the `#fetch` method to raise an error if the required parameter is missing.
+
+```ruby
+class ThingController < ApplicationController
+  # ... actions
+
+  private
+
+   def an_important_param
+     params.fetch(:important_and_required)
+   end
+end
+```
+
+
+## But why?
+
+When you access an item from the `params` like I am above it implies that the value is pretty important for the correct functioning of your action.
```

Comment:
*Needing to access an item from the `params` collection implies that the value is important for the correct functioning of your action.